neusomatic/python/filter_candidates.py

```diff
@@ -259,25 +261,57 @@ def filter_candidates(candidate_record):
                 final_records.append([chrom, pos - 1, ref, alt, line])
         final_records = sorted(final_records, key=lambda x: x[0:2])
         if dbsnp:
-            filtered_bed = pybedtools.BedTool(map(lambda x:
-                                                  pybedtools.Interval(x[1][0], int(x[1][1]),
-                                                                      int(x[1][
-                                                                          1]) + 1,
-                                                                      x[1][2], x[1][3], str(x[0])),
-                                                  enumerate(final_records))).sort()
-            dbsnp = pybedtools.BedTool(dbsnp).each(lambda x:
-                                                   pybedtools.Interval(x[0], int(x[1]),
-                                                                       int(x[
-                                                                           1]) + 1,
-                                                                       x[3], x[4])).sort()
-            non_in_dbsnp_1 = filtered_bed.window(dbsnp, w=0, v=True)
-            non_in_dbsnp_2 = filtered_bed.window(dbsnp, w=0).filter(
-                lambda x: x[1] != x[7] or x[3] != x[9] or x[4] != x[10]).sort()
+            filtered_bed = tempfile.NamedTemporaryFile(
+                prefix="tmpbed_", suffix=".bed", delete=False)
+            filtered_bed = filtered_bed.name
+            with open(filtered_bed, "w") as f_o:
+                for x in enumerate(final_records):
+                    f_o.write("\t".join(map(str, [x[1][0], int(x[1][1]), int(
+                        x[1][1]) + 1, x[1][2], x[1][3], str(x[0])])) + "\n")
+            cmd = "bedtools sort -i {}".format(filtered_bed)
+            filtered_bed = run_bedtools_cmd(cmd, run_logger=thread_logger)
+
+            dbsnp_tmp = tempfile.NamedTemporaryFile(
+                prefix="tmpbed_", suffix=".bed", delete=False)
+            dbsnp_tmp = dbsnp_tmp.name
+            with open(dbsnp_tmp, "w") as f_o:
+                with open(dbsnp, "r") as f_i:
+                    for line in f_i:
```

Comment:
@msahraeian same here. Try to move some of this general code to a function which reads intervals or generates BED from VCF.

Reply:
@marghoob I added multiple functions to make it more modular, please check.